src/_blog/metrics-in-go-libp2p.md

```diff
@@ -0,0 +1,162 @@
+
+## How users can enable metrics
+
+Metrics have been enabled by default from go-libp2p [v0.26.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.26.0). All you need to do is setup a Prometheus exporter for the collected metrics. 
+
+```go
+
+func main() {
+        http.Handle("/metrics", promhttp.Handler())
+	go func() {
+		http.ListenAndServe(":2112", nil) // Any port is fine
+	}()
+
+	host, err := libp2p.New()
+        // err handling
+        ...
+}
+```
+Now just point your prometheus instance to scrape from `:2122/metrics`
+
+By default, metrics are sent to the default prometheus Registerer. To use a different Registerer from the default prometheus registerer, use the option `libp2p.PrometheusRegisterer`. 
+
+```go
+
+func main() {
+	reg := prometheus.NewRegistry()
+        http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
+	go func() {
+		http.ListenAndServe(":2112", nil) // Any port is fine
+	}()
+
+	host, err := libp2p.New(
+                libp2p.PrometheusRegisterer(reg),
+        )
+        // err handling
+        ...
+}
+```
+
+<!-- TODO: incorporate this PR: https://github.com/libp2p/go-libp2p/pull/2232 -->
+
+### Discovering what metrics are available
+
+go-libp2p provides metrics and grafana dashboards for all its major subsystems out of the box. You can check https://github.com/libp2p/go-libp2p/tree/master/dashboards for the grafana dashboards available. Another great way to discover available metrics is to open prometheus ui and type `libp2p_(libp2p-package-name)_` and find available metrics from autocomplete. For Ex: `libp2p_autonat_` gives you the list of all metrics exported from [AutoNAT](https://github.com/libp2p/specs/tree/master/autonat).
+
+<div class="container" style="display:flex; column-gap:10px; justify-content: center; align-items: center;">
+    <figure>
+        <img src="../assets/metrics-in-go-libp2p-prometheus-ui.png" width="750">
+        <figcaption style="font-size:x-small;">
+                EvtLocalAddressesUpdated
+        </figcaption>
+    </figure>
+</div>
+
+
+## How are metrics useful?  
+
+I'll share two cases where having metrics were extremely helpful for us in go-libp2p. One case deals with being able to debug a memory leak and one where adding two new metrics helped us with development of a new feature. 
+
+### Debugging with metrics
+
+We were excited about adding metrics because it gave us the opportunity to observe exactly what was happening within the system. One of the first system we added metrics to was the Event Bus. When we added event bus metrics, we were immediately able to see discrepancy between two of our metrics, `EvtLocalReachabilityChanged` and `EvtLocalAddressesUpdated`. You can see the details on the [github issue](https://github.com/libp2p/go-libp2p/issues/2046)
```

Comment:
```suggestion
We were excited about adding metrics because it gave us the opportunity to observe exactly what was happening within the system. One of the first systems we added metrics to was the Event Bus. 
The event bus is used to pass event notifications between different libp2p components.
When we added event bus metrics, we were immediately able to see discrepancy between two of our metrics, `EvtLocalReachabilityChanged` and `EvtLocalAddressesUpdated`. You can see the details on the [GitHub issue](https://github.com/libp2p/go-libp2p/issues/2046)
```

Reply:
using this.